microsoft/testsuites/sysbench/sysbenchperf.py

```diff
@@ -0,0 +1,406 @@
+
+            log.debug(res)
+
+            self._send_subtest_msg(
+                result,
+                res.name,
+                res.status,
+                res.message,
+            )
+
+        if len(failure) > 0:
+            log.debug(f"Failed count: {len(failure)}")
+            fail(f"Sysbench-fileio test failed: {failure}")
+
+    @TestCaseMetadata(
+        description="""
+            Runs Sysbench test for memory
+        """,
+        priority=3,
+    )
+    def verify_sysbench_memory_perf(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+        variables: Dict[str, Any],
+    ) -> None:
+        memory_operation: List[str] = ["read", "write"]
+        memory_access_mode: List[str] = ["seq", "rnd"]
+        failure = []
+        sysbench = node.tools[Sysbench]
+
+        for op in memory_operation:
+            for access_mode in memory_access_mode:
+                subtest_name: str = f"{access_mode}_{op}_default_memory_perf_check"
+                res: SysbenchTestResult = SysbenchTestResult()
+                output: ExecutableResult = sysbench.run_memory_perf(
+                    memory_access_mode=access_mode,
+                    memory_oper=op,
+                )
+
+                if output.exit_code != 0:
+                    failure.append(f"{output.cmd}")
+                    res.name = subtest_name
+                    res.message = (
+                        f"{subtest_name} failed with " f"exit code {output.exit_code}"
+                    )
+                    res.status = TestStatus.FAILED
+                else:
+                    res = process_memory_perf_result(
+                        output.stdout,
+                        subtest_name,
+                    )
+
+                self._send_subtest_msg(
+                    result,
+                    res.name,
+                    res.status,
+                    res.message,
+                )
+
+        if len(failure) > 0:
+            fail(f"Sysbench-memory test failed: {failure}")
+
+    def _send_subtest_msg(
+        self,
+        test_result: TestResult,
+        test_name: str,
+        test_status: TestStatus,
+        test_message: str = "",
+    ) -> None:
+        subtest_msg = send_sub_test_result_message(
+            test_result=test_result,
+            test_case_name=test_name,
+            test_status=test_status,
+            test_message=test_message,
+        )
+
+        notifier.notify(subtest_msg)
+
+
+def process_perf_result(data: str, testcase_name: str) -> SysbenchTestResult:
+    non_debug_pattern = r"^(?!.*DEBUG: ).*$"
+    debug_patern = r".*DEBUG: .*"
+
+    result: Dict[str, Any] = {}
+
+    # Find all non-DEBUG lines in the text
+    non_debug_text = "\n".join(
+        re.findall(
+            non_debug_pattern,
+            data,
+            re.MULTILINE,
+        )
+    )
+    debug_text = "\n".join(
```

Comment:
Why does it need to differentiate debug text?

Reply:
We have min/max/avg metrics for latency and debug parts (when we run with debug=on)

So reg-ex to get the min/max/avg, we can re-use if we split up the debug text. I added the sample output for reference now.

Reply:
Do you mean reuse the output? The patterns are different, so they won't conflict. For example, the non-debug test, you can write it like `r"(?P<metric>min|avg|max|95th percentile|sum):\s+(?P<value>[\d.]+)\n"`, the debug text, you can use like `r"(?P<metric>min|avg|max):\s+(?P<value>[\d.]+)s"`, and capture the events separately.

Reply:
How about this comment? Is it fixed? The non-debug logic is not necessary too.

Reply:
We have removed non-debug now. We dont need to push verbose metrics.

Reply:
Please remove the `non_debug_pattern = r"^(?!.*DEBUG: ).*$"` also.

Reply:
Done